api/tests/unit/application/users/user-controller_test.js

```diff
@@ -37,75 +37,94 @@ describe('Unit | Controller | user-controller', function () {
+                    email: 'john.dodoe@example.net',
+                    cgu: true,
+                    password,
+                  },
                 },
               },
             },
-          },
-          hFake
-        );
+            hFake
+          );
 
-        // then
-        expect(userSerializer.serialize).to.have.been.calledWith(savedUser);
-        expect(response.source).to.deep.equal(expectedSerializedUser);
-        expect(response.statusCode).to.equal(201);
+          // then
+          expect(userSerializer.serialize).to.have.been.calledWith(savedUser);
+          expect(response.source).to.deep.equal(expectedSerializedUser);
+          expect(response.statusCode).to.equal(201);
+        });
       });
 
-      it('should call the user creation usecase', async function () {
-        // given
-        const useCaseParameters = {
-          user: deserializedUser,
-          password,
-          localeFromHeader,
-          campaignCode: null,
-        };
-
-        // when
-        await userController.save(
-          {
-            payload: {
-              data: {
-                attributes: {
-                  'first-name': 'John',
-                  'last-name': 'DoDoe',
-                  email: 'john.dodoe@example.net',
-                  cgu: true,
-                  password,
+      describe('when there is a locale cookie', function () {
+        it('should return a serialized user with "locale" attribute and a 201 status code', async function () {
+          // given
+          const localeFromCookie = 'fr-FR';
+          const expectedSerializedUser = { message: 'serialized user', locale: localeFromCookie };
+          const savedUser = new User({ email, locale: localeFromCookie });
+
+          const useCaseParameters = {
+            user: { ...deserializedUser, locale: localeFromCookie },
+            password,
+            localeFromHeader,
+            campaignCode: null,
+          };
+
+          userSerializer.serialize.returns(expectedSerializedUser);
+          createUserUsecaseStub.resolves(savedUser);
+
+          // when
+          const response = await userController.save(
+            {
+              payload: {
+                data: {
+                  attributes: {
+                    'first-name': 'John',
+                    'last-name': 'DoDoe',
+                    email: 'john.dodoe@example.net',
+                    cgu: true,
+                    password,
+                  },
                 },
               },
+              state: {
+                locale: localeFromCookie,
+              },
             },
-          },
-          hFake
-        );
-
-        // then
-        expect(usecases.createUser).to.have.been.calledWith(useCaseParameters);
+            hFake
+          );
+
+          // then
+          expect(usecases.createUser).to.have.been.calledWith(useCaseParameters);
+          expect(userSerializer.serialize).to.have.been.calledWith(savedUser);
```

Comment:
je pense qu'on peut supprimer ce expect(userSerializer.serialize).to.have.been.calledWith(savedUser);  
les seralizer étant normalement testé en TU
ou alors supprimer expect(response.source).to.deep.equal(expectedSerializedUser); ?

Reply:
c'est enlevé 😄